libraries/node-core-library/src/Executable.ts

```diff
@@ -113,11 +114,221 @@ interface IExecutableContext {
   windowsExecutableExtensions: string[];
 }
 
-interface ICommandLineFixup {
+interface ICommandLineOptions {
   path: string;
   args: string[];
 }
 
+/**
+ * Process information sourced from the system. This process info is sourced differently depending
+ * on the operating system:
+ * - On Windows, this uses the `wmic.exe` utility.
+ * - On Unix, this uses the `ps` utility.
+ *
+ * @public
+ */
+export interface IProcessInfo {
+  /**
+   * The name of the process.
+   *
+   * @remarks On Windows, the process name will be empty if the process is a kernel process.
+   * On Unix, the process name will be empty if the process is the root process.
+   */
+  processName: string;
+  /**
+   * The process ID.
+   */
+  processId: number;
+  /**
+   * The parent process info.
+   *
+   * @remarks On Windows, the parent process info will be undefined if the process is a kernel process.
+   * On Unix, the parent process info will be undefined if the process is the root process.
+   */
+  parentProcessInfo?: IProcessInfo;
+
+  /**
+   * The child process infos.
+   */
+  childProcessInfos: IProcessInfo[];
+}
+
+// Match the newline character(s) at the end of a line of text in order to split on it. On Windows,
+// it is possible for multiple \r characters may precede the \n character, so we need to match all
+// of them.
+const NEWLINE_REGEX: RegExp = /\r*\n/;
+
+async function* readLinesFromStreamAsync(stream: NodeJS.ReadableStream): AsyncGenerator<string> {
+  let remaining: string = '';
+  for await (const chunk of stream) {
+    remaining += chunk;
+    // eslint-disable-next-line @rushstack/no-new-null
+    let match: RegExpMatchArray | null;
+    while ((match = remaining.match(NEWLINE_REGEX)) && match.index !== undefined && match.index >= 0) {
```

Comment:
I think this will be simpler if you just use indexOf for `\n` and then trim off all `/[\r\n]/` from returned lines.

Reply:
Moved to `Text` class and it makes more sense to use the included regexes there. However, I'm now using `matches` to avoid multiple calls to the regex.